src/site/content/en/blog/building-a-sidenav-component/index.md

```diff
@@ -0,0 +1,333 @@
+to ensure the `box-shadow` of the sidenav doesn't peek into the main viewport when it's hidden.
+
+```css
+@media (max-width: 540px) {
+  #sidenav-open {
+    visibility: hidden;
+    transform: translateX(-110vw);
+    will-change: transform;
+    transition: 
+      transform var(--duration) var(--easeOutExpo),
+      visibility 0s linear var(--duration);
+  }
+}
+```
+
+##### Sidenav in
+
+When the `#sidenav` element matches as `:target`, set the `translateX()` position to homebase `0`, 
+and watch as CSS slides the element from its out position of `-110vw`, to its "in" 
+position of `0` over `var(--duration)` when the URL hash is changed. 
+
+```css
+@media (max-width: 540px) {
+  #sidenav-open:target {
+    visibility: visible;
+    transform: translateX(0);
+    transition: 
+      transform var(--duration) var(--easeOutExpo);
+  }
+}
+```
+
+#### Transition visibility
+
+The goal now is to hide the menu from screenreaders when it's out, 
+so systems don't put focus into an offscreen menu. I accomplish this by setting a 
+visibility transition when the `:target` changes.
+
+- When going in, don't transition visibility; be visible right away so I can see the element slide in and accept focus. 
+- When going out, transition visibility but delay it, so it flips to `hidden` at the end of the transition out.
+
+### Accessibility UX enhancements {: #ux-enhancements }
+
+#### Links
+
+This solution relies on changing the URL in order for the state to be managed. 
+Naturally, the `<a>` element should be used here, and it gets some nice accessibility 
+features for free. Let's adorn our interactive elements with labels clearly articulating intent.
+
+```html
+<a href="#" id="sidenav-close" title="Close Menu" aria-label="Close Menu"></a>
+
+<a href="#sidenav-open" id="sidenav-button" class="hamburger" title="Open Menu" aria-label="Open Menu">
+  <svg>...</svg>
+</a>
+```
+
+<figure class="w-figure w-figure--fullbleed">
+  <video playsinline controls autoplay loop muted class="w-screenshot">
+    <source src="https://storage.googleapis.com/atoms-sandbox.google.com.a.appspot.com/keyboard-voiceover.mp4">
+  </video>
+  <figcaption class="w-figure">
+    A demo of the voiceover and keyboard interaction UX.
+  </figcaption>
+</figure>
+
+Now our primary interaction buttons clearly state their intent for both mouse and keyboard.
+
+#### `:is(:hover, :focus)`
+
+This handy CSS functional pseudo-selector lets us swiftly be inclusive 
+with our hover styles by sharing them with focus as well.
+
+```css
+.hamburger:is(:hover, :focus) svg > line {
+  stroke: hsl(var(--brandHSL));
+}
+```
+
+#### Sprinkle on Javascript 
+
+##### Press `escape` to close
+
+The `Escape` key on your keyboard should close the menu right? Let's wire that up.
+
+```js
+const sidenav = document.querySelector('#sidenav-open')
+
+sidenav.addEventListener('keyup', event => {
+  if (event.code === 'Escape')
+    document.location.hash = ''
+})
```

Comment:
```suggestion
const sidenav = document.querySelector('#sidenav-open');

sidenav.addEventListener('keyup', event => {
  if (event.code === 'Escape') document.location.hash = '';
});
```

* https://google.github.io/styleguide/jsguide.html#formatting-braces-all
* https://google.github.io/styleguide/jsguide.html#formatting-semicolons-are-required

Can you update the JS in your Glitch to also follow the style guide? I know that these things can turn into debates but the fact is we're on a Google site, publishing code associated to Google's brand, so we should follow the Google JS style guide.

Reply:
np